### PR TITLE
CAL-229: Addressed false positive OWASP vulnerabilties

### PR DESCRIPTION
**Pull request by brianfelix**
#### What does this PR do?
Address OWASP vulnerabilities by suppressing false positives using dependency-check-maven-config.xml.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@clockard @gordocanchola @vinamartin @mcalcote 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@shaundmorris
@stustison
#### How should this be tested?
Full OWASP build should pass
#### What are the relevant tickets?
[CAL-229](https://codice.atlassian.net/browse/CAL-229)
#### Checklist:
- [ ] Documentation Updated
	- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by brianfelix** [comment]
@shaundmorris Geowebcache vulnerabilities will be investigated but for the time being will be left in the dependency-check-maven-config.xml as CVE suppressions. Documentation is being added to let users know that geowebcache should not be installed in a secured environment.

File: dependency-check-maven-config.xml [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+            which does not apply here.
+        </notes>
+        <cve>CVE-2007-1085</cve>
+    </suppress>
+
+    <suppress>
+        <notes>
+            CVE-2007-3150 is a JavaScript injection vulnerability related to Google Desktop which
+            does
+            not apply here.
```
Comment (mcalcote) [inline dependency-check-maven-config.xml]: odd spacing?

File: dependency-check-maven-config.xml [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+            Spring, so marking it as a false positive.
+        </notes>
+        <cve>CVE-2011-2730</cve>
+    </suppress>
+
+    <suppress>
+        <notes>
+            CVE-2011-5034: Applies to
+            org.apache.servicemix.specs.activation-api-1.1-2.5.0.jar/META-INF/maven/org.apache.geronimo.specs/geronimo-activation_1.1_spec/pom.xml
+            ServiceMix embeds some Specs provided by Geronimo but does not use any of the effected
```
Comment (mcalcote) [inline dependency-check-maven-config.xml]: affected? also can probably keep `libraries.` on the same line here considering the length of the line above it

File: dependency-check-maven-config.xml [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+        <notes>
+            CVE-2016-1000031: Applies to commons-fileupload-1.2.1, suppressing because the
+            vulnerable class DiskFileItem is not used in the project
+        </notes>
+        <cve>CVE-2016-1000031</cve>
+    </suppress>
+
+    <suppress>
+        <notes><![CDATA[
+        false positive the effected camel version is 2.12 this uses a later version and does use the XSLT component
```
Comment (mcalcote) [inline dependency-check-maven-config.xml]: Notes above are written in sentences with capitalization/punctuation. Not a requirement, just noticed.

File: dependency-check-maven-config.xml [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+
+
+    <suppress>
+        <notes><![CDATA[
+                Suppressing vulnerabilities CVE-2013-4221 and CVE-2013-4221 as the offending jar file (org.restlet-2.1.1.jar)
+        is being manually removed from the Solr War and replaced with the fixed version. These should be removed when
+        Solr is updated (DDF-1110). See pom file for details.
+   file name: solr-4.7.2.war: org.restlet-2.1.1.jar
+   ]]></notes>
+        <sha1>E12C23B962C925F2681729AFA1E40066A350AD27</sha1>
```
Comment (vinamartin) [inline dependency-check-maven-config.xml]: Are you sure you want the specific sha here and below?

File: dependency-check-maven-config.xml [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+        false positive the effected camel version is 2.12 this uses a later version and does use the XSLT component
+   file name: proxy-camel-servlet-2.9.0-SNAPSHOT.jar
+   ]]></notes>
+        <cve>CVE-2014-0003</cve>
+    </suppress>
+
+    <suppress>
+        <notes><![CDATA[
+        this CVE is generating a lot of false positives it should only include jackson-dataformat-xml jar
+         but it is catching all jackson dependencies unfortunately I can't fix it so if we start depending
```
Comment (mcalcote) [inline dependency-check-maven-config.xml]: I think you should remove `unfortunately I can't fix it`.

File: dependency-check-maven-config.xml [hunk 6]
```diff
@@ -0,0 +1,485 @@
...
+        <notes><![CDATA[
+   file name: solr-*.jar
+   OWASP is getting confused by our version number being on a jar with solr in the name we are on solr 6.0+ which is not affected by this issue.
+   ]]></notes>
+        <cve>CVE-2012-6612</cve>
+    </suppress>
+
+    <suppress>
+        <notes>
+            CVE-2014-0050: Applies to commons-fileupload-1.2.1, suppressing due to replacing jar
```
Comment (bdeining) [inline dependency-check-maven-config.xml]: Curious why some of these have `<![CDATA[` tags and some don't?

File: dependency-check-maven-config.xml [hunk 7]
```diff
@@ -0,0 +1,485 @@
...
+            ServiceMix embeds some Specs provided by Geronimo but does not use any of the effected
+            libraries.
+        </notes>
+        <cve>CVE-2011-5034</cve>
+    </suppress>
+
+
+    <suppress>
+        <notes><![CDATA[
+                Suppressing vulnerabilities CVE-2013-4221 and CVE-2013-4221 as the offending jar file (org.restlet-2.1.1.jar)
```
Comment (gordocanchola) [inline dependency-check-maven-config.xml]: Should the second CVE-2013-4221 be CVE-2013-4271?
Should we just remove the reference to the 4271 and copy the notes there as well?
That way each suppression has documented notes.

File: dependency-check-maven-config.xml [hunk 8]
```diff
@@ -0,0 +1,485 @@
...
+        is being manually removed from the Solr War and replaced with the fixed version. These should be removed when
+        Solr is updated (DDF-1110). See pom file for details.
+   file name: solr-4.7.2.war: org.restlet-2.1.1.jar
+   ]]></notes>
+        <sha1>E12C23B962C925F2681729AFA1E40066A350AD27</sha1>
+        <cve>CVE-2013-4221</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[
```
Comment (gordocanchola) [inline dependency-check-maven-config.xml]: Looking at this suppression alone doesn't provide any information as to why it's being suppressed.
Should we just copy the notes from 4221?

File: dependency-check-maven-config.xml [hunk 9]
```diff
@@ -0,0 +1,485 @@
...
+
+    <suppress>
+        <notes><![CDATA[
+        false positive CVE is unrelated
+   file name: org.apache.servicemix.bundles.not-yet-commons-ssl
+   ]]></notes>
+        <cve>CVE-2004-0009</cve>
+    </suppress>
+
+    <!-- these are the geowebcache vulnerabilities it is not installed by default and it is only experimental these security issues would need to be resolved before geowebcache can be installed in a production enviroment-->
```
Comment (gordocanchola) [inline dependency-check-maven-config.xml]: Should we put this comment in the notes field of each of the geowebache suppressions?

File: dependency-check-maven-config.xml (hunk 6 above)
Comment (brianfelix) [inline dependency-check-maven-config.xml]: Fixed. All notes now have the CDATA tag.

File: dependency-check-maven-config.xml (hunk 5 above)
Comment (brianfelix) [inline dependency-check-maven-config.xml]: Fixed: Removed.

File: dependency-check-maven-config.xml (hunk 4 above)
Comment (brianfelix) [inline dependency-check-maven-config.xml]: Fixed: Removed.

File: dependency-check-maven-config.xml (hunk 7 above)
Comment (brianfelix) [inline dependency-check-maven-config.xml]: Fixed.

File: dependency-check-maven-config.xml (hunk 8 above)
Comment (brianfelix) [inline dependency-check-maven-config.xml]: Fixed

File: dependency-check-maven-config.xml (hunk 9 above)
Comment (brianfelix) [inline dependency-check-maven-config.xml]: Fixed: Added that comment to each geowebcache suppression.

File: dependency-check-maven-config.xml (hunk 2 above)
Comment (brianfelix) [inline dependency-check-maven-config.xml]: Fixed.

File: dependency-check-maven-config.xml (hunk 1 above)
Comment (brianfelix) [inline dependency-check-maven-config.xml]: Fixed.

File: dependency-check-maven-config.xml (hunk 3 above)
Comment (brianfelix) [inline dependency-check-maven-config.xml]: Fixed.

**Comment by shaundmorris** [comment]
Merging. This file will only affect builds that are using OWASP dependency-check.

**Pull request merged**